ui/app/src/components/CreateFolderDialog/CreateFolderContainer.tsx

```diff
@@ -86,15 +87,21 @@ export function CreateFolderContainer(props: CreateFolderContainerProps) {
   };
 
   const onCreateFolder = (site: string, path: string, name: string) => {
-    createFolder(site, path, name).subscribe({
-      next() {
-        onCreated?.({ path, name, rename });
-        dispatch(updateCreateFolderDialog({ isSubmitting: false, hasPendingChanges: false }));
-      },
-      error(response) {
-        dispatch(
-          batchActions([showErrorDialog({ error: response }), updateCreateFolderDialog({ isSubmitting: false })])
-        );
+    fetchSandboxItem(site, `${path}/${name}`).subscribe((item) => {
+      if (item) {
+        dispatch(batchActions([fetchSandboxItemComplete({ item }), updateCreateFolderDialog({ isSubmitting: false })]));
+      } else {
+        createFolder(site, path, name).subscribe({
```

Comment:
```ts
fetch().pipe(
  tap(item && dispatch(fetchComplete()),
  filter(!item),
  switchMap(createFile())
).subscribe(...)
```
 Apply pattern to both dialogs.

Make sure the return of the fetch is nullish now that the services doesn't 404 and such.